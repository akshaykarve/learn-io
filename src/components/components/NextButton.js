import React from 'react';
import '../ComponentStyle.css';

const NextButton=(props)=>{
	let but;
	if(props.nextPlatforms===false){
		but=<button disabled style={{color:'grey'}} className='paginationButtons'>Next</button>
	}else{
			but=<button className='paginationButtons' onClick={()=>{props.setSkip(props.skip+props.limit)}}>Next</button>
	}
	return but;
}

export default NextButton;