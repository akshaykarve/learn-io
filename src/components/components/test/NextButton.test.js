import React from 'react';
import renderer from 'react-test-renderer';
import {default as NextButton} from "../NextButton";

describe("Next Button Component", ()=>{
	it('not platforms for next page', () => {
		
		const tree = renderer.create(
			<NextButton nextPlatforms={false}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('some platforms for next page', () => {
		
		const tree = renderer.create(
			<NextButton nextPlatforms={true}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});