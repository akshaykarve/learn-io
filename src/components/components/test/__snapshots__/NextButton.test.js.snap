// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Next Button Component not platforms for next page 1`] = `
<button
  className="homeButton"
  disabled={true}
  style={
    Object {
      "color": "grey",
    }
  }
>
  Next
</button>
`;

exports[`Next Button Component some platforms for next page 1`] = `
<button
  className="homeButton"
  onClick={[Function]}
>
  Next
</button>
`;
